### PR TITLE
Add a BDO copy method to IBaseDataObjectHelper.

### PR DESCRIPTION
**Pull request by jdcove2**
This PR adds a method to IBaseDataObjectHelper that will shallow copy the attributes of a BaseDataObject to another BaseDataObject.

### CODE REVIEW

File: src/main/java/emissary/core/IBaseDataObjectHelper.java [hunk 1]
```diff
@@ -97,6 +98,115 @@ public static IBaseDataObject clone(final IBaseDataObject iBaseDataObject, final
...
+     * call would not be necessary.
+     * 
+     * @param bdo the BaseDataObject to set the field on.
+     * @param fieldName the name of the field to be set.
+     * @param object the object that the field is to be set to.
+     * @throws IllegalAccessException if this {@code Field} object is enforcing Java language access control and the
+     *         underlying field is either inaccessible or final.
+     * @throws NoSuchFieldException if a field with the specified name is not found.
+     */
+    public static void setPrivateFieldValue(final BaseDataObject bdo, final String fieldName, @Nullable final Object object)
```
Comment (drivenflywheel) [inline src/main/java/emissary/core/IBaseDataObjectHelper.java:199]: Why would this need to be `public`?

File: src/main/java/emissary/core/BaseDataObject.java [hunk 2]
```diff
@@ -754,10 +757,14 @@ public String printMeta() {
 
     @Override
     public void addProcessingError(final String err) {
-        if (this.procError == null) {
-            this.procError = new StringBuilder();
+        if (err == null) {
+            this.procError = null;
```
Comment (drivenflywheel) [inline src/main/java/emissary/core/BaseDataObject.java:761]: Why would an "add" method remove existing entries?

Comment (drivenflywheel) [reply src/main/java/emissary/core/BaseDataObject.java:761]: > What suggestion do you have for resolving your example above?

Revert the `addProcessingError` change.  If your `copy` use case needs a `clearProcessingErrors` method, create it. Don't add unexpected new behavior to the existing method just to satisfy the new edge case.

**Comment by jpdahlke** [comment]
closing for now, this can be reopened if this becomes a need in the future

**Pull request closed without merging**